mashumaro/core/meta/code/builder.py

```diff
@@ -554,16 +554,36 @@ def _unpack_method_set_value(
                 could_be_none=False if could_be_none else True,
             )
         )
-        if unpacked_value != "value":
-            self.add_line(f"value = d.get('{alias or fname}', MISSING)")
-            packed_value = "value"
-        elif has_default:
-            self.add_line(f"value = d.get('{alias or fname}', MISSING)")
-            packed_value = "value"
+        if self.get_config().loose_deserialization:
+            if unpacked_value != "value":
+                self.add_line(
+                    f"value = d.get('{alias}', d.get('{fname}', MISSING))"
+                )
+                packed_value = "value"
+            elif has_default:
+                self.add_line(
+                    f"value = d.get('{alias}', d.get('{fname}', MISSING))"
+                )
+                packed_value = "value"
+            else:
+                self.add_line(
+                    f"__{fname} = d.get('{alias}', d.get('{fname}', MISSING))"
+                )
+                packed_value = f"__{fname}"
+                unpacked_value = packed_value
```

Comment:
`d.get(..., d.get(..., MISSING))` will always be leading to two key population operations. It will be more performant to  generate the following:
```python
value = d.get('{alias}', MISSING)
if value is MISSING:
    value = d.get('{fname}', MISSING)
```